userapi/src/main/java/fk/prof/userapi/api/ProfileStoreAPIImpl.java

```diff
@@ -19,54 +20,51 @@
 import java.time.temporal.ChronoUnit;
 import java.util.*;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.stream.Collectors;
 
 /**
- * Interacts with the {@link AsyncStorage} based on invocations from controller
+ * Interacts with the {@link AsyncStorage} and the {@link ClusterAwareCache} based on invocations from controller.
  * Created by rohit.patiyal on 19/01/17.
  */
 public class ProfileStoreAPIImpl implements ProfileStoreAPI {
-    private static Logger LOGGER = LoggerFactory.getLogger(ProfileStoreAPIImpl.class);
 
     private static final String DELIMITER = "/";
-    private static final String WORKER_POOL_NAME = "aggregation.loader.pool";
     private static final String VERSION = "v0001";
-    private final int profileLoadTimeout;
 
-    private Vertx vertx;
-    private AsyncStorage asyncStorage;
-    private AggregatedProfileLoader profileLoader;
+    private final AsyncStorage asyncStorage;
+    private final AggregatedProfileLoader profileLoader;
 
-    private WorkerExecutor workerExecutor;
+    private final WorkerExecutor workerExecutor;
+    private final Vertx vertx;
+    private final int loadTimeout;
 
-    private Cache<String, AggregatedProfileInfo> cache;
     private Cache<String, AggregationWindowSummary> summaryCache;
 
-    /* stores all requested futures that are waiting on file to be loaded from S3. If a file loadFromInputStream
+    private final ClusterAwareCache clusterAwareCache;
+
+    /* stores all requested futures that are waiting on file to be loaded from S3. If a file loading
     * is in progress, this map will contain its corresponding key */
-    private Map<String, FuturesList<Object>> futuresForLoadingFiles;
+    private final Map<String, FuturesList<AggregationWindowSummary>> filesBeingLoaded;
 
-    public ProfileStoreAPIImpl(Vertx vertx, AsyncStorage asyncStorage, int maxIdleRetentionInMin, Integer profileLoadTimeout, Integer workerPoolSize) {
-        this.vertx = vertx;
+    public ProfileStoreAPIImpl(Vertx vertx, AsyncStorage asyncStorage, ClusterAwareCache clusterAwareCache, Configuration config) {
         this.asyncStorage = asyncStorage;
         this.profileLoader = new AggregatedProfileLoader(this.asyncStorage);
```

Comment:
aggregatedprofileloader can be injected to this class from userapimanager, the same way it is passed to clustercache rather than instantiating it here. asyncstorage is required for listing of app/cluster/proc etc. While today, same storage is used to load profiles, that may not be the case later when metadata and data can come from different stores.

Reply:
Done.